### PR TITLE
heimdal: use `berkeley-db@5` and clarify `flex` runtime dep

### PR DESCRIPTION
**Pull request by triallax**
See #100796.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/heimdal.rb [hunk 1]
```diff
@@ -24,8 +24,7 @@ class Heimdal < Formula
   keg_only "conflicts with Kerberos"
 
   depends_on "bison" => :build
-  depends_on "berkeley-db"
-  depends_on "flex"
+  depends_on "flex" => :build
```
Comment (carlocab) [inline Formula/heimdal.rb:27]: Linkage is broken, so this is needed at runtime.

Comment (mhmdanas) [reply Formula/heimdal.rb:27]: Ah, I had the feeling this was actually needed. Any suggested comments to clarify this?

Comment (carlocab) [reply Formula/heimdal.rb:27]: `# required at runtime`?

**Comment by mhmdanas** [comment]
Strange, on Linux there is an `libotp.so` links with `libdb`, even though I thought passing `--without-berkeley-db` to `./configure` would've been sufficient...

**Pull request closed without merging**